### PR TITLE
fix: Fix Timestamp, Duration and FieldMask marshaling in REST transport

### LINKED ISSUE #333: Proto JSON cannonical encoding is not supported for special types
With an exception of `int64<->string` encoding (implemented in https://github.com/googleapis/proto-plus-python/pull/267) the rest of special types [JSON Mapping ](https://developers.google.com/protocol-buffers/docs/proto3#json) are not supported. Most commonly failing types are `Timestamp`, `FieldMask` and `Duration`

This is crucial to support rest transport in Python.

### PR DESCRIPTION
**Pull request by vam-google**
This fixes https://github.com/googleapis/proto-plus-python/issues/333

BEGIN_COMMIT_OVERRIDE
fix: Fix Timestamp, Duration and FieldMask marshaling in REST transport
Release-As: 1.22.0
END_COMMIT_OVERRIDE

### CODE REVIEW

File: proto/marshal/rules/field_mask.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+class FieldMaskRule:
+    """A marshal between FieldMask and strings.
+
+    See https://github.com/googleapis/proto-plus-python/issues/333
+    and
+    https://developers.google.com/protocol-buffers/docs/proto3#json
+    for more details.
+    """
+
+    def to_python(self, value, *, absent: bool = None):
```
Comment (atulep) [inline proto/marshal/rules/field_mask.py:27]: Is it expected to be a string ?

File: proto/marshal/rules/field_mask.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return value
+
+    def to_proto(self, value):
```
Comment (atulep) [inline proto/marshal/rules/field_mask.py:30]: What are possible types for value ?

Comment (atulep) [reply proto/marshal/rules/field_mask.py:30]: Is this protoplus proto or protobuf proto ?

**Review by atulep — APPROVED** [review]

File: proto/marshal/rules/field_mask.py (hunk 1 above)
Comment (atulep) [inline proto/marshal/rules/field_mask.py:27]: This is not used - should be removed ?

Comment (vam-google) [reply proto/marshal/rules/field_mask.py:27]: what specifically? This just follows a similar convention of other Rules (they do exaclty the same thing as I have here).

File: proto/marshal/rules/field_mask.py (hunk 2 above)
Comment (vam-google) [reply proto/marshal/rules/field_mask.py:30]: Proto-plus

File: proto/marshal/rules/field_mask.py (hunk 1 above)
Comment (vam-google) [reply proto/marshal/rules/field_mask.py:27]: It is not used, but other rules still have it (i ti snot used in most places), so keeping it like this for consistency.

**Pull request merged**